### DISCUSSION TITLE
TICKER_HANDLER not being reset after @typeclass/force self

### DISCUSSION THREAD

**Discussion by RodRitter** [evennia/evennia#2678] [Evennia Questions] [answered]
Good day all,

I've implemented a `TICKER_HANDLER` timer on the `at_create_object` part of my character. Initially, I set it to 2 seconds to debug and test, but now I can't remove it or update the timing.

I've reset my character multiple times (`@typeclass/force self`) and `reset` the server, but I can't seem to remove or update the ticker.

#### Before
```
    def at_object_creation(self):
       ....
        TICKER_HANDLER.add(2, self.natural_healing)
```
#### After
```
    def at_object_creation(self):
       ....
        self.msg("at_object_creation changed v2")
        TICKER_HANDLER.remove(2, self.natural_healing) <- I even tried this
        TICKER_HANDLER.add(60, self.natural_healing)
```
```
> @typeclass/force self
> at_object_creation changed v2
> reset
```
The ticker is still running every 2 seconds.

Any help is appreciated

**Comment by RodRitter** [comment, marked as answer]
Had to dig into the `ticker_handler` code but managed to find a way to display all of them. Turns out there was a stray ticker of 3 seconds that I missed for some reason.

Is there an admin command for showing active tickers on an object?

**Reply by Griatch** [reply]
If you think about this practically, there is no way for Evennia to know about the 2-second ticker you had in that `at_object_creation` hook before you commented it out. Once the ticker is created there is no record of _where_ it's created and no way for the system to know that when you rerun the `at_object_creation` hook, there is now _no_ such ticker defined. You must clear that yourself.

If you use Evennia 1.0-dev (develop branch) there is a `tickers` command for seeing the currently running tickers. This is just a convenience tool at this time though - you still need to stop it manually in code.

**Reply by RodRitter** [reply]
Thanks, yea I'm still trying to understand how it works, so I was a bit confused. I figured out I had to remove it in the code, but I missed a ticker I created before.
